### PR TITLE
Add swatch attributes

### PR DESCRIPTION
**Pull request by piotrgrundas**
I want to merge this change because it adds swatch attributes.

[Saleor PR](https://github.com/mirumee/saleor/pull/7261)

**PR intended to be tested with API branch:**  feature/swatch-attribute 

### Screenshots
imageimage
image
image

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `[data-test-id]` are added for new elements
4. [x] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3092 Swatch attributes](https://app.clickup.com/t/2549495/SALEOR-3092)

File: src/attributes/components/AttributePage/AttributePage.tsx [hunk 1]
```diff
@@ -54,6 +54,7 @@ export interface AttributePageProps {
   };
   onNextPage: () => void;
   onPreviousPage: () => void;
+  children: (data: AttributePageFormData) => React.ReactNode;
```
Comment (piotrgrundas) [inline src/attributes/components/AttributePage/AttributePage.tsx:57]: I need it to have info about selected attr inputType at creation 👀

**Review by bmigirl — CHANGES_REQUESTED** [review]

File: src/attributes/components/AttributeSwatchField/messages.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { defineMessages } from "react-intl";
+
+export const defined = defineMessages({
```
[the remark this answers is not in the document]
Comment (piotrgrundas) [reply src/attributes/components/AttributeSwatchField/messages.ts]: Why do u review outdated code? 🤔

File: src/attributes/components/AttributeSwatchField/AttributeSwatchField.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+              description: "swatch attribute color picker label"
+            }),
+            value: "picker"
+          },
+          {
+            label: formatMessage({
+              defaultMessage: "Image",
+              description: "swatch attribute image label"
+            }),
+            value: "image"
```
Comment (orzechdev) [inline src/attributes/components/AttributeSwatchField/AttributeSwatchField.tsx:93]: According to convention, move messages to `./messages.ts`

File: src/attributes/components/AttributeValueEditDialog/AttributeValueEditDialog.tsx [hunk 4]
```diff
@@ -39,14 +40,24 @@ const AttributeValueEditDialog: React.FC<AttributeValueEditDialogProps> = ({
...
-    name: maybe(() => attributeValue.name, "")
+    name: attributeValue?.name ?? "",
+    ...(attributeValue?.fileUrl
+      ? {
+          fileUrl: attributeValue?.fileUrl,
+          contentType: attributeValue?.contentType
+        }
+      : {
+          value: attributeValue?.value ?? ""
+        })
```
Comment (orzechdev) [inline src/attributes/components/AttributeValueEditDialog/AttributeValueEditDialog.tsx:56]: To make it more readable, maybe make it like:
```ts
const attributeValueFields = attributeValue?.fileUrl
      ? {
          fileUrl: attributeValue?.fileUrl,
          contentType: attributeValue?.contentType
        }
      : {
          value: attributeValue?.value ?? ""
        }

const initialForm: AttributeValueEditDialogFormData = {
  name: attributeValue?.name ?? "",
  ...attributeValueFields,
}

```

File: src/attributes/utils/data.ts [hunk 5]
```diff
@@ -61,6 +81,22 @@ function getSimpleAttributeData(
...
+  values: AttributeValueEditDialogFormData[]
+) {
+  return {
+    ...data,
+    metadata: undefined,
+    privateMetadata: undefined,
+    storefrontSearchPosition: parseInt(data.storefrontSearchPosition, 10),
+    values: values.map(({ fileUrl, name, value, contentType }) => ({
+      name,
+      ...(fileUrl ? { fileUrl, contentType } : { value })
```
Comment (orzechdev) [inline src/attributes/utils/data.ts:95]: Same here, what do you think about making util for it e.g. `getAttributeValueTypeFields(attributeValue)`?

**Review by orzechdev — CHANGES_REQUESTED** [review]

File: src/attributes/components/AttributeSwatchField/AttributeSwatchField.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  const { formatMessage } = useIntl();
+  const classes = useStyles();
+  const [processing, setProcessing] = useState(false);
+  const [uploadFile] = useFileUploadMutation({});
+  const [type, setType] = useState<SwatchType>(
+    data.fileUrl ? "image" : "picker"
+  );
+
+  const handleColorChange = (hex: string) =>
+    set({ value: hex, fileUrl: undefined, contentType: undefined });
```
[the remark this answers is not in the document]
Comment (piotrgrundas) [reply src/attributes/components/AttributeSwatchField/AttributeSwatchField.tsx:47]: API type.

File: src/attributes/components/AttributeValues/AttributeValues.tsx [hunk 7]
```diff
@@ -103,6 +117,14 @@ const AttributeValues: React.FC<AttributeValuesProps> = ({
         <TableHead>
           <TableRow>
             <TableCell className={classes.columnDrag} />
+            {isSwatch && (
+              <TableCell className={classes.columnSwatch}>
+                <FormattedMessage
```
[the remark this answers is not in the document]
Comment (piotrgrundas) [reply src/attributes/components/AttributeValues/AttributeValues.tsx:122]: Not one message, nor style has a separate file here 🙄 ...

**Review by orzechdev — APPROVED** [review]

**Review by pwgryglak — APPROVED** [review]

**Review by bmigirl — APPROVED** [review]

**Review by pwgryglak — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**